### PR TITLE
Update 3.3-di-changes.rst

### PR DESCRIPTION
**Pull request by DonCallisto**
Related to https://github.com/symfony/symfony-docs/pull/9676#pullrequestreview-115232548

### CODE REVIEW

File: service_container/3.3-di-changes.rst [hunk 1]
```diff
@@ -567,6 +567,12 @@ Start by updating the service ids to class names:
     classes (e.g. ``Twig_Extensions_Extension_Intl`` instead of ``Twig\Extensions\IntlExtension``),
     you can't redefine the service as ``Twig_Extensions_Extension_Intl: ~`` and
     you must keep the original ``class`` parameter.
+    
+.. caution::
+
+    If a service is processed by `Compiler Pass`_ you could face a  "You have requested a non-existent service" error.
```
Comment (xabbuh) [inline service_container/3.3-di-changes.rst]: We usually split lines after the first word that crosses the 72nd character.

File: service_container/3.3-di-changes.rst [hunk 2]
```diff
@@ -567,6 +567,12 @@ Start by updating the service ids to class names:
     classes (e.g. ``Twig_Extensions_Extension_Intl`` instead of ``Twig\Extensions\IntlExtension``),
     you can't redefine the service as ``Twig_Extensions_Extension_Intl: ~`` and
     you must keep the original ``class`` parameter.
+    
+.. caution::
+
+    If a service is processed by `Compiler Pass`_ you could face a  "You have requested a non-existent service" error. 
+    To get rid of this, be sure that Compiler Pass is using ``findDefinition`` instead of ``getDefinition`` as the latter
```
Comment (xabbuh) [inline service_container/3.3-di-changes.rst]: I would use `findDefinition()` and `getDefinition()` (i.e. include the parentheses)

File: service_container/3.3-di-changes.rst [hunk 3]
```diff
@@ -778,3 +784,5 @@ can be autowired. The final configuration looks like this:
 You can now take advantage of the new features going forward.
 
 .. _`FrameworkExtension for 3.3.0`: https://github.com/symfony/symfony/blob/7938fdeceb03cc1df277a249cf3da70f0b50eb98/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php#L247-L284
+
+.. _`Compiler Pass`: https://symfony.com/doc/current/service_container/compiler_passes.html
```
Comment (xabbuh) [inline service_container/3.3-di-changes.rst]: we should use the `doc` role instead:

```rst
If a service is processed by a :doc:`compiler pass </service_container/compiler_passes>` [...]

File: service_container/3.3-di-changes.rst (hunk 2 above)
Comment (xabbuh) [reply service_container/3.3-di-changes.rst]: We could also add a note that it is generally a good idea to check with `has()` first if there actually is an alias or definition we are looking for.

**Comment by DonCallisto** [comment]
@xabbuh thank you for the review. I've made changes in a separate commit so you can check only the diff.
Moreover I would like to point out that, for consistency reason, even 

```
.. _`FrameworkExtension for 3.3.0`: https://github.com/symfony/symfony/blob/7938fdeceb03cc1df277a249cf3da70f0b50eb98/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php#L247-L284
```

should be converted to `:doc:` reference.

WDYT?

File: service_container/3.3-di-changes.rst [hunk 4]
```diff
@@ -568,6 +568,16 @@ Start by updating the service ids to class names:
     you can't redefine the service as ``Twig_Extensions_Extension_Intl: ~`` and
     you must keep the original ``class`` parameter.
 
+.. caution::
+
+    If a service is processed by :doc:`compiler pass </service_container/compiler_passes>`
```
Comment (xabbuh) [inline service_container/3.3-di-changes.rst]: minor typo: [...] by a [...]

Comment (xabbuh) [reply service_container/3.3-di-changes.rst]: and there should be a comma at the end of the line

File: service_container/3.3-di-changes.rst [hunk 5]
```diff
@@ -568,6 +568,16 @@ Start by updating the service ids to class names:
...
+    you could face a  "You have requested a non-existent service" error.
+    To get rid of this, be sure that Compiler Pass is using ``findDefinition()``
```
Comment (xabbuh) [inline service_container/3.3-di-changes.rst]: [...] that the compiler pass is [...]

File: service_container/3.3-di-changes.rst [hunk 6]
```diff
@@ -568,6 +568,16 @@ Start by updating the service ids to class names:
...
+    instead of ``getDefinition()`` as the latter won't take aliases into
```
Comment (xabbuh) [inline service_container/3.3-di-changes.rst]: [...] `getDefinition()`. The latter won't [...]

File: service_container/3.3-di-changes.rst [hunk 7]
```diff
@@ -568,6 +568,16 @@ Start by updating the service ids to class names:
...
+    account when looking up for services.
+    Furthermore is always recommendend to check for definition existance
```
Comment (xabbuh) [inline service_container/3.3-di-changes.rst]: [...] it is [...]

Comment (xabbuh) [reply service_container/3.3-di-changes.rst]: [...] existence using `has()`.

**Comment by xabbuh** [comment]
The `doc` role does not work for external links (for example, to link to code on GitHub). We can only use it for cross-references inside this documentation.

**Comment by DonCallisto** [comment]
Right, didn't see that was an external link.

**Comment by DonCallisto** [comment]
At the end will you squash commits or do you prefer me to do the squash?

**Comment by xabbuh** [comment]
Whatever you prefer :) As long as you do not push a merge commit we can easily squash your commits while merging.

**Comment by DonCallisto** [comment]
Ok, I have no preferences, just let me know if I need to perform other actions on this.

**Review by xabbuh — APPROVED** [review]

**Comment by javiereguiluz** [comment]
Thank you Samuele.

**Pull request closed without merging**